runner/pool/pool.go

```diff
@@ -701,11 +713,44 @@ func (r *basePoolManager) AddRunner(ctx context.Context, poolID string, aditiona
 		AditionalLabels:   aditionalLabels,
 	}
 
-	_, err = r.store.CreateInstance(r.ctx, poolID, createParams)
+	instance, err := r.store.CreateInstance(r.ctx, poolID, createParams)
 	if err != nil {
 		return errors.Wrap(err, "creating instance")
 	}
 
+	defer func() {
+		if err != nil && instance.ID != "" {
+			if err := r.ForceDeleteRunner(instance); err != nil {
+				r.log("failed to cleanup instance: %s", instance.Name)
+			}
+		}
+	}()
+
+	labels := r.getLabelsForInstance(pool)
+	// Attempt to create JIT config
+	jitConfig, runner, err := r.helper.GetJITConfig(ctx, instance, pool, labels)
+	if err == nil {
```

Comment:
@maigl This should fall back to the old way of registering a runner. Any error in creating a JIT config should make it fall back to the old registration method.